scripts/inject-deterministic.py

```diff
@@ -25,26 +115,44 @@ def response(self, flow):
 
                 with open("scripts/catapult/deterministic.js", "r") as jsfile:
                     js = jsfile.read().replace("REPLACE_LOAD_TIMESTAMP", str(millis))
+                    js = re.sub(re.compile("//.*?\n"), "", js)
+
                     if js not in html:
-                        scriptIndex = re.search('(?i).*?<head.*?>', html)
-                        if scriptIndex is None:
-                            scriptIndex = re.search('(?i).*?<html.*?>', html)
-                        if scriptIndex is None:
-                            scriptIndex = re.search('(?i).*?<!doctype html>', html)
-                        if scriptIndex is None:
+                        script_index = re.search('(?i).*?<head.*?>', html)
+                        if script_index is None:
+                            script_index = re.search('(?i).*?<html.*?>', html)
+                        if script_index is None:
+                            script_index = re.search('(?i).*?<!doctype html>', html)
+                        if script_index is None:
                             ctx.log.info("No start tags found in request {}. Skip injecting".format(flow.request.url))
                             return
+                        script_index = script_index.end()
 
-                        scriptIndex = scriptIndex.end()
+                        nonce = None
 
-                        new_html = html[:scriptIndex] + "<script>" + js + "</script>" + html[scriptIndex:]
+                        if flow.response.headers.get("Content-Security-Policy", False):
+                            nonce = self.get_nonce_from_headers(flow.response.headers)
+                            ctx.log.info("nonce : %s" % nonce)
 
+                            if not nonce:
```

Comment:
If I understand correctly, we only need to do this if we have a `script-src` directive. I would suggest the following tweak:

```python
if self.get_csp_script_sources(flow.response.headers) and not nonce:
    ...
```